lib/metrics.py

```diff
@@ -202,6 +204,48 @@ def calculate_nrmse(filename1, filename2, width, height, nframes = 1, fourcc = "
     RawFile(filename2, width, height, nframes, fourcc),
     nframes, __compare_nrmse)
 
+def BD_RATE(R1, PSNR1, R2, PSNR2, piecewise=0):
+  lR1 = np.log(R1)
+  lR2 = np.log(R2)
+
+  # rate method
+  p1 = np.polyfit(PSNR1, lR1, 3)
+  p2 = np.polyfit(PSNR2, lR2, 3)
+
+  # integration interval
+  min_int = max(min(PSNR1), min(PSNR2))
+  max_int = min(max(PSNR1), max(PSNR2))
+
+  # find integral
+  if piecewise == 0:
+      p_int1 = np.polyint(p1)
+      p_int2 = np.polyint(p2)
+
+      int1 = np.polyval(p_int1, max_int) - np.polyval(p_int1, min_int)
+      int2 = np.polyval(p_int2, max_int) - np.polyval(p_int2, min_int)
+  else:
+      lin = np.linspace(min_int, max_int, num=100, retstep=True)
+      interval = lin[1]
+      samples = lin[0]
+      v1 = scipy.interpolate.pchip_interpolate(np.sort(PSNR1), lR1[np.argsort(PSNR1)], samples)
+      v2 = scipy.interpolate.pchip_interpolate(np.sort(PSNR2), lR2[np.argsort(PSNR2)], samples)
+      # Calculate the integral using the trapezoid method on the samples.
+      int1 = np.trapz(v1, dx=interval)
+      int2 = np.trapz(v2, dx=interval)
+
+  # find avg diff
+  avg_exp_diff = (int2-int1)/(max_int-min_int)
+  avg_diff = (np.exp(avg_exp_diff)-1)*100
+  return avg_diff
```

Comment:
This looks like an exact copy of the code from https://pypi.org/project/bd-metric/.  It is probably better to add it as a dependency instead of copying the code.